components/application-broker/internal/broker/provision.go

```diff
@@ -267,20 +271,23 @@ func (svc *ProvisionService) do(inputParams map[string]interface{}, iID internal
 		return
 	}
 
-	// persist Knative Subscription
-	if err := svc.persistKnativeSubscription(appName, appSvcID, ns); err != nil {
-		svc.log.Printf("Error persisting Knative Subscription: %v", err)
-		opDesc := fmt.Sprintf("provisioning failed while persisting Knative Subscription for application: %s namespace: %s on error: %s", appName, ns, err)
-		svc.updateStateFailed(iID, opID, opDesc)
-		return
-	}
+	if !svc.newEventingFlow {
 
-	// enable the namespace default Knative Broker
-	if err := svc.enableDefaultKnativeBroker(ns); err != nil {
-		opDesc := fmt.Sprintf("provisioning failed while enabling default Knative Broker for namespace: %s"+
-			" on error: %s", ns, err)
-		svc.updateStateFailed(iID, opID, opDesc)
-		return
+		// persist Knative Subscription
+		if err := svc.persistKnativeSubscription(appName, appSvcID, ns); err != nil {
+			svc.log.Printf("Error persisting Knative Subscription: %v", err)
+			opDesc := fmt.Sprintf("provisioning failed while persisting Knative Subscription for application: %s namespace: %s on error: %s", appName, ns, err)
+			svc.updateStateFailed(iID, opID, opDesc)
+			return
+		}
+
+		// enable the namespace default Knative Broker
+		if err := svc.enableDefaultKnativeBroker(ns); err != nil {
+			opDesc := fmt.Sprintf("provisioning failed while enabling default Knative Broker for namespace: %s"+
+				" on error: %s", ns, err)
+			svc.updateStateFailed(iID, opID, opDesc)
+			return
+		}
```

Comment:
Do we still need to execute `createOrUpdatePeerAuthentication` in the new flow?

If not, then please also move it inside the `if !svc.newEventingFlow` statement.

Reply:
I'm not sure
@sayanh can you advise, please?

Reply:
removed as advised by @sayanh